core/src/main/java/com/datastrato/gravitino/utils/NameIdentifierUtil.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.utils;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.exceptions.IllegalNameIdentifierException;
+import com.datastrato.gravitino.exceptions.IllegalNamespaceException;
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+/**
+ * A name identifier is a sequence of names separated by dots. It's used to identify a metalake, a
+ * catalog, a schema or a table. For example, "metalake1" can represent a metalake,
+ * "metalake1.catalog1" can represent a catalog, "metalake1.catalog1.schema1" can represent a
+ * schema.
+ */
+public class NameIdentifierUtil {
+
+  /**
```

Comment:
A private constructor here is preferable

Reply:
> A private constructor here is preferable

Done